<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:macro name="arduino_ros2_control" params="prefix">
    <ros2_control name="${prefix}arduino_diff_drive" type="system">
      <hardware>
        <plugin>arduino_hardware_interface/ArduinoSystemHardware</plugin>
        <param name="port" value="/dev/ttyACM0"/>
        <param name="baudrate" value="115200"/>
      </hardware>
      <joint name="left_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="right_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
    </ros2_control>
  </xacro:macro>
</robot>


